python/ray/async_compat.py

```diff
@@ -11,3 +16,81 @@ def sync_to_async(func):
         return func(*args, **kwargs)
 
     return wrapper
+
+
+# Class encapsulate the get result from direct actor.
```

Comment:
Can we make two separate classes instead of overloading? More clear and looks like no overlap here

Reply:
I think two classes are a bit too complicated. Because we are essentially doing the following:
```python
if isinstance(obj,ray.ObjectID):
    ...
else:
    ...
```
but it's less clear. So the namedtuple here is just to document where the object comes from.